ui/src/test/java/de/ipb_halle/lbac/admission/LdapAdmissionSubSystemTest.java

```diff
@@ -0,0 +1,169 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package de.ipb_halle.lbac.admission;
+
+import de.ipb_halle.lbac.admission.mock.LdapHelperMock;
+import de.ipb_halle.lbac.base.TestBase;
+import static de.ipb_halle.lbac.base.TestBase.prepareDeployment;
+import de.ipb_halle.lbac.service.InfoObjectService;
+import java.util.Set;
+import javax.inject.Inject;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+
+/**
+ *
+ * @author fmauz
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(Arquillian.class)
+public class LdapAdmissionSubSystemTest extends TestBase {
+
+    private static final long serialVersionUID = 1L;
+
+    @Inject
+    private UserBeanMock userBean;
+
+    @Inject
+    private LdapProperties ldapProperties;
+
+    LdapAdmissionSubSystem system;
+    LdapHelperMock ldapHelper;
+
+    @Before
+    public final void init() {
+        userBean = new UserBeanMock();
+        userBean.setCurrentAccount(publicUser);
+        userBean.setLdabProperties(ldapProperties);
+        userBean.setNodeService(nodeService);
+        userBean.setMemberService(memberService);
+        userBean.setMemberShipService(membershipService);
+        userBean.setGlobalAdmissionContext(context);
+
+        ldapHelper = new LdapHelperMock();
+        system = new LdapAdmissionSubSystem(ldapHelper);
+        ldapProperties.LdapBasicsInit();
+    }
+
+    @After
+    public void finish() {
+        entityManagerService.doSqlUpdate("DELETE FROM info");
+
+    }
+
+    @Test
+    public void test001_authenticate_withoutLdapEntry() {
+        Assert.assertFalse(system.authenticate(publicUser, "admin", userBean));
+    }
+
+    @Test
+    public void test002_authenticate_withoutLdapEnabled() {
+        entityManagerService.doSqlUpdate("INSERT INTO info(key,value) VALUES('LDAP_ENABLE','false')");
+        Assert.assertFalse(system.authenticate(publicUser, "admin", userBean));
+    }
+
+    @Test
+    public void test003_authenticate_withoutExistingLdabUser() {
+        entityManagerService.doSqlUpdate("INSERT INTO info(key,value) VALUES('LDAP_ENABLE','true')");
+        ldapHelper.addLdapObject("CN=ldab_user_name", "ldab_email", "ldab_login", "ldac_user_name_edited", "+xxx", MemberType.USER, "uniqueId-001");
+        Assert.assertFalse(system.authenticate(publicUser, "admin", userBean));
+    }
+
+    @Test
+    public void test004_authenticate_withExistingLBACUser() {
```

Comment:
Comment what happens here (e.g. the user name was changed in the LDAP and we expect this to change in our DB)